java/ql/src/experimental/Security/CWE/CWE-200/AndroidFileIntentSource.qll

```diff
@@ -0,0 +1,116 @@
+/** Provides summary models relating to file content inputs of Android. */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.frameworks.android.Android
+
+/** The `startActivityForResult` method of Android `Activity`. */
+class StartActivityForResultMethod extends Method {
+  StartActivityForResultMethod() {
+    this.getDeclaringType().getASupertype*() instanceof AndroidActivity and
+    this.getName() = "startActivityForResult"
+  }
+}
+
+/** Android class instance of `GET_CONTENT` intent. */
+class GetContentIntent extends ClassInstanceExpr {
+  GetContentIntent() {
+    this.getConstructedType() instanceof TypeIntent and
+    this.getArgument(0).(CompileTimeConstantExpr).getStringValue() =
+      "android.intent.action.GET_CONTENT"
+    or
+    exists(Field f |
+      this.getArgument(0) = f.getAnAccess() and
+      f.hasName("ACTION_GET_CONTENT") and
+      f.getDeclaringType() instanceof TypeIntent
+    )
+  }
+}
+
+/** Android intent data model in the new CSV format. */
+private class AndroidIntentDataModel extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "android.content;Intent;true;addCategory;;;Argument[-1];ReturnValue;taint",
+        "android.content;Intent;true;addFlags;;;Argument[-1];ReturnValue;taint",
```

Comment:
`addFlags` not used?

Reply:
Thanks @smowton for fixing the test issue. I have thought to move newly added methods/constants to the end of the <code>Intent</code> file then noticed you already fixed the issue.

Yes, <code>addFlags</code> is not being used in the test program. I thought it would be helpful to add it for completeness. You can decide to remove it or not.